scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py

```diff
@@ -185,3 +188,24 @@ def _determine_chopper_mode(ws):
             return 'PDF', polaris_advanced_config.pdf_focused_cropping_values
     else:
         raise ValueError("Chopper frequency not in log data. Please specify a chopper mode")
+
+
+def fast_fourier_filter(ws, freq_params=None):
+    if not freq_params:
+        return
+    # This is a simple fourier filter using the FFTSmooth to get a WS with only the low radius components, then
+    # subtracting that from the merged WS
+
+    x_range = ws.dataX(0)
+    # The param p in FFTSmooth defined such that if the input ws has N bins then in the fourier space ws it will cut of
+    # all frequencies in bins N/p and above. this is calculated by pi/(k*dx) when k is the cutoff frequency desired.
+    lower_freq_param = round(np.pi / (freq_params[0] * (x_range[1] - x_range[0])))
+    # this if giving the FFTSmooth the data in the form of S(Q)-1, later we use PDFFourierTransform with Q(S(Q)-1)
+    # it does not matter which we use in this case.
```

Comment:
I managed to reproduce this formula in the end so that's all good. Please can you add these two facts to the comment so it's a bit less cryptic where the formula for lower_freq_param comes from:
- upper limit of spectrum in Fourier space is 1/(2 * bin width in Q space)
- the Fourier transform in FFTSmooth uses sin(2*pi*Qr) whereas PDFFourierTransform uses sin(Qr) so additional factor of 2 * pi required
Also "this if giving the..." needs to be "this is giving the..."